### PR TITLE
User story 3.1

### LINKED ISSUE #3: User Story 3
Image

As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the names of the top 5 customers
who have conducted the largest number of successful transactions with my merchant
And next to each customer name I see the number of successful transactions they have
conducted with my merchant

### PR DESCRIPTION
**Pull request by dani-wilson**
# Describe the changes below:
add tests, new test data(thanks @westonio 😆), model method, css, and views for merchant dashboard

# Relevant user story:
Close #3 
As a merchant,
When I visit my merchant dashboard (/merchants/:merchant_id/dashboard)
Then I see the names of the top 5 customers
who have conducted the largest number of successful transactions with my merchant
And next to each customer name I see the number of successful transactions they have
conducted with my merchant

# Specific feedback requests? Include file name and line:
There's one test that isn't passing, but it isn't mine, and I'm not sure how that happened...

# Before submitting, check the following:
- [] Entire test suite is passing - one is not, but it isn't mine and I'm not sure what to do about it
- [] SimpleCov test percentage - 98.11%

# Note any co-authors / anything else you'd like to add:
sorry this took so long :\

### CODE REVIEW

File: app/assets/stylesheets/application.css [hunk 1]
```diff
@@ -195,3 +195,30 @@ ul {
...
+
+.item-sections-header-container th, td {
+  border: none !important;
+  background-color: white;
+  color: black;
+}
+
+.table_headers {
+  background-color: #e1e1e1 !important;
+}
```
Comment (westonio) [inline app/assets/stylesheets/application.css]: Yay CSS!! 👍

File: app/models/merchant.rb [hunk 2]
```diff
@@ -6,4 +6,9 @@ class Merchant < ApplicationRecord
   def invoices
     Invoice.joins(items: :merchant).where(items: { merchant: self }).distinct
   end
+
+  def favorite_customers
+    ids = invoices.pluck(:id)   
+    Customer.select("first_name, last_name, count(result)").joins(invoices: :transactions).where(transactions: {result: "success"}).where(invoices: {id: ids}).order(count: :desc).group("first_name, last_name").limit(5)
+  end
```
Comment (westonio) [inline app/models/merchant.rb:15]: This looks good! 
If you find it helpful in the future, you can start a new line before each command for better visibility.

E.g. 
```
Customer.select("first_name, last_name, count(result)")
  .joins(invoices: :transactions)
  .where(transactions: {result: "success"})
  .where(invoices: {id: ids})
  .order(count: :desc).group("first_name, last_name")
  .limit(5)
```

File: spec/models/merchant_spec.rb [hunk 3]
```diff
@@ -35,4 +35,71 @@
...
+      @transaction_13 = @invoice_7.transactions.create!(credit_card_number: 4084466070588807, credit_card_expiration_date: "04/27", result: "success")
+      @transaction_14 = @invoice_8.transactions.create!(credit_card_number: 4993984512460266, credit_card_expiration_date: "04/27", result: "success")
+      @transaction_15 = @invoice_8.transactions.create!(credit_card_number: 4993984512460266, credit_card_expiration_date: "04/27", result: "failed")
+      @transaction_16 = @invoice_9.transactions.create!(credit_card_number: 4504301557459341, credit_card_expiration_date: "04/27", result: "success")
+      @transaction_17 = @invoice_9.transactions.create!(credit_card_number: 4504301557459341, credit_card_expiration_date: "04/27", result: "success")
+      @transaction_17 = @invoice_9.transactions.create!(credit_card_number: 4504301557459341, credit_card_expiration_date: "04/27", result: "success")
+      @transaction_17 = @invoice_9.transactions.create!(credit_card_number: 4504301557459341, credit_card_expiration_date: "04/27", result: "success")
```
Comment (westonio) [inline spec/models/merchant_spec.rb:102]: Nice data sets to test your new methods/ views with!

**Review by westonio — APPROVED** [review]
Nice job :)

File: app/models/merchant.rb (hunk 2 above)
Comment (awiley33) [reply app/models/merchant.rb:15]: Wow thanks @westonio , I didn't know that! I do like the visibility here

File: spec/features/dashboard/index_spec.rb [hunk 4]
```diff
@@ -2,25 +2,87 @@
...
     expect(page).to have_link("My Invoices")
     expect(page).to_not have_link("Admin")
   end
+  #user story 3
+  it "displays the names of the top 5 customers" do
+    visit "/merchants/#{@merchant_1.id}/dashboard"
+
+    expect(page).to have_content("Favorite Customers")
+    expect(page).to have_content(@merchant_1.favorite_customers.first.first_name)
+  end
```
Comment (awiley33) [inline spec/features/dashboard/index_spec.rb:87]: testing looks great!

**Review by awiley33 — APPROVED** [review]
Good work Dani!

**Pull request merged**